// ngx-drag-scroll/esm2015/lib/ngx-drag-scroll.module.js
import { NgModule } from '@angular/core';
import { DragScrollComponent } from './ngx-drag-scroll.component';
import { DragScrollItemDirective } from './ngx-drag-scroll-item';
import * as ɵngcc0 from '@angular/core';
export class DragScrollModule {
}
DragScrollModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DragScrollModule });
DragScrollModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function DragScrollModule_Factory(t) { return new (t || DragScrollModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DragScrollModule, { declarations: function () { return [DragScrollComponent,
        DragScrollItemDirective]; }, exports: function () { return [DragScrollComponent,
        DragScrollItemDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DragScrollModule, [{
        type: NgModule,
        args: [{
                exports: [
                    DragScrollComponent,
                    DragScrollItemDirective
                ],
                declarations: [
                    DragScrollComponent,
                    DragScrollItemDirective
                ]
            }]
    }], null, null); })();
