// ngx-drag-scroll/esm2015/lib/ngx-drag-scroll-item.js
import { Directive, ElementRef, Input, HostBinding, Inject } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class DragScrollItemDirective {
    constructor(elementRef) {
        this.display = 'inline-block';
        this._dragDisabled = false;
        this._elementRef = elementRef;
    }
    get dragDisabled() { return this._dragDisabled; }
    set dragDisabled(value) { this._dragDisabled = value; }
}
DragScrollItemDirective.ɵfac = function DragScrollItemDirective_Factory(t) { return new (t || DragScrollItemDirective)(ɵngcc0.ɵɵdirectiveInject(ElementRef)); };
DragScrollItemDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: DragScrollItemDirective, selectors: [["", "drag-scroll-item", ""]], hostVars: 2, hostBindings: function DragScrollItemDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("display", ctx.display);
    } }, inputs: { dragDisabled: ["drag-disabled", "dragDisabled"] } });
DragScrollItemDirective.ctorParameters = () => [
    { type: ElementRef, decorators: [{ type: Inject, args: [ElementRef,] }] }
];
DragScrollItemDirective.propDecorators = {
    display: [{ type: HostBinding, args: ['style.display',] }],
    dragDisabled: [{ type: Input, args: ['drag-disabled',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DragScrollItemDirective, [{
        type: Directive,
        args: [{
                selector: '[drag-scroll-item]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef, decorators: [{
                type: Inject,
                args: [ElementRef]
            }] }]; }, { display: [{
            type: HostBinding,
            args: ['style.display']
        }], dragDisabled: [{
            type: Input,
            args: ['drag-disabled']
        }] }); })();
